ee/vellum_ee/workflows/display/tests/workflow_serialization/test_basic_inline_prompt_node_serialization.py

```diff
@@ -274,3 +274,49 @@ def test_serialize_workflow():
         "name": "BasicInlinePromptWithFunctionsWorkflow",
         "module": ["tests", "workflows", "basic_inline_prompt_node_with_functions", "workflow"],
     }
+
+
+def test_serialize_workflow_with_descriptor_functions():
+    """Test that serialization handles BaseDescriptor instances in functions list."""
+    from vellum import ChatMessagePromptBlock, JinjaPromptBlock
+    from vellum.workflows import BaseWorkflow
+    from vellum.workflows.inputs import BaseInputs
+    from vellum.workflows.nodes import InlinePromptNode
+    from vellum.workflows.state import BaseState
+    from vellum_ee.workflows.display.workflows.get_vellum_workflow_display_class import get_workflow_display
+
+    class TestInputs(BaseInputs):
+        noun: str
+
+    from vellum.workflows.references.node import NodeReference
+
+    class MockMCPClientNode:
+        class Outputs:
+            tools = NodeReference(
+                name="tools",
+                types=(list,),
+                instance=["tools.0", "tools.1"],  # Non-callable objects that will cause the error
+                node_class=object,
+            )
+
+    class TestInlinePromptNodeWithDescriptorFunctions(InlinePromptNode):
+        ml_model = "gpt-4o"
+        blocks = [
+            ChatMessagePromptBlock(
+                chat_role="SYSTEM",
+                blocks=[JinjaPromptBlock(template="Test {{noun}}")],
+            ),
+        ]
+        prompt_inputs = {"noun": TestInputs.noun}
+        functions = MockMCPClientNode.Outputs.tools
+
+    class TestWorkflow(BaseWorkflow[TestInputs, BaseState]):
+        graph = TestInlinePromptNodeWithDescriptorFunctions
+
+    workflow_display = get_workflow_display(workflow_class=TestWorkflow)
+
+    serialized = workflow_display.serialize()
+
+    assert "workflow_raw_data" in serialized
+    assert "input_variables" in serialized
+    assert "output_variables" in serialized
```

Comment:
Instead of these assertions, find the prompt node in workflow_raw_data and validate the shape of the `functions` attribute